docs/discussion.md

```diff
@@ -0,0 +1,205 @@
+# **Interpolation**
+
+## **How the btwxt library works**
+## Big Ladder Software
+
+## 1-D Linear
+We do not know the true function $f$. We only know values of $f$ at a set of grid points $x_i$.
+
+To find $f(x)$ for $x$ in $(x_0, x_1)$,
+
+let $\mu=(x-x_0)/(x_1-x_0)$.
+
+$f(x)=(1-\mu)\cdot f(x_0) + \mu \cdot f(x_1)$
+
+
+<p align="center">
+    <img width="70%" src="./figs/fig_1Dlinear.png"> 
+</p>
+
+## 2-D Linear
+We would like to extend the 1-D interpolation to 2-D.
+
+$$\begin{align*}
+f(\mu)=&(1-\mu_1)\cdot [(1-\mu_0)\cdot f(x_{00}) + \mu_0 \cdot f(x_{10})]\\
++&\mu_1\cdot [(1-\mu_{0})\cdot f(x_{01}) + \mu_0 \cdot f(x_{11})]
+\end{align*}$$
+
+Or, to do the calculation in one pass:
+
+$$\begin{align*}
+f(\mu)=&(1-\mu_0)\cdot (1-\mu_1)\cdot f(x_{00}) + \mu_0\cdot (1-\mu_1)\cdot f(x_{10})\\
++&(1-\mu_0)\cdot \mu_1\cdot f(x_{01}) + \mu_0 \cdot \mu_1\cdot f(x_{11})
+\end{align*}$$
+
+
+<p align="center">
+    <img width="70%" src="./figs/fig_2Dlinear.png"> 
+</p>
+
+Four independent terms, one for each vertex.
+
+## 2-D Linear pseudocode
```

Comment:
Actually I think the 2-D linear pseudocode is both distracting and anyway you'd unroll the loop if you wrote it as real code, not think of it in this conditional format. Maybe we remove this?